bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
     */
         HttpURLConnection conn = null;
         String serviceURL = "";
-           
+
         try {
-	    if (incomingBinaryPOSTData.length == 0) return XML_DEC + "<results>" + STATUS_OK + "<transcript />" + "</results>";
+            if (incomingBinaryPOSTData.length == 0) {
+                return XML_DEC + "<results>" + STATUS_OK + "<transcript />" + "</results>";
+            }
 
             // 'VCAP_SERVICES' contains all the details of services bound to this application
-	    JsonReader jsonReader = Json.createReader(new StringReader(System.getenv("VCAP_SERVICES")));
-	    JsonObject serviceInfo = jsonReader.readObject();
-	    jsonReader.close();
+            JsonObject credentials = readJsonObject(new StringReader(System.getenv("VCAP_SERVICES")))
+                                    .getJsonArray(SERVICE_NAME)
+                                    .getJsonObject(0)
+                                    .getJsonObject("credentials");
 
-            // Get the Service Credentials for SpeechToText
-	    JsonObject credentials = serviceInfo.getJsonArray(SERVICE_NAME)
-		.getJsonObject(0).getJsonObject("credentials");
 
-            try {
-                serviceURL = credentials.getString("url");
-            } catch (Exception e) {}
-            // If we didn't find a URL for the service in VCAP_SERVICES,
-            // use the original.
+
+            serviceURL = credentials.getString("url");
+
```

Comment:
Yeah, the "correct" method is that JsonObject queries should be able to return Optionals instead of throwing. The equivalent Haskell is just:

`result fallbackURL = maybe fallbackURL id . readMaybe

I think the idea with Java8 Optionals though is that they were supposed to replace Nulls, so we probably won't actually see a rework of libraries to add these methods in Java9 or whatever.